### PR TITLE
Scan: Jetpack site with no Scan purchase is shown a loading placeholder indefinitely

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

Visiting the _Scan_ section of a Jetpack site without Scan enabled shows the loading placeholder indefinitely instead of showing the upsell section (see screenshot below). This PR fixes this.

### Implementation notes

There are several things to not here:
- The `JETPACK_SCAN_REQUEST_FAILURE` action needed to be explicitly dispatched from the error handler.
- For the Scan use case, the component `UpsellSwitch` used `state.jetpackScan.scan` for the loading logic, while it should have used `state.jetpackScan.requestStatus` (`state.jetpackScan.scan` stays an empty object if the request fails and the loading state cannot be identified).
- `UpsellSwitch` is shared between Scan and Backup sections.
- The loading logic in `UpsellSwitch` had code specific to the Backup section. With the new update, it made sense to extract that logic in a function passed as a prop, to make the component section agnostic.
- The scan information is fetched in several parts of the app. This could make the Scan section reload completely (showing the loading state) while the customer was interacting with it. The `UpsellSwitch` now makes sure to render the loading state only once while mounted.

### Testing instructions

Before testing, make sure you've got 2 Jetpack sites available. One should have Backup but no Scan, and the other one the opposite. The goal here is to make sure the appropriate sections load for Backup and Scan.

### Switching sections

For each site:

1. Go to a page that's neither _Backup_ nor _Scan_, let's say _Plan_
2. Go to _Backup_. Verify that you see the Backup feature or upsell
3. Refresh the page and check again
4. Go to _Scan_. Verify that you see the Scan feature or upsell
5. Refresh the page and check again

### Switching site

1. Go to _Scan_ with one of your site and check that you see what's expected
2. Switch to the other site and check you see what's expected with this site
3. Repeat for backup

### Screenshots

Screen Shot 2020-06-22 at 3 59 17 PM

### CODE REVIEW

**Comment by monsieur-z** [comment]
Thanks for the feedback @rcanepa. I've been caught up in the paths I used for developing. The issues you mentioned should be fixed, but before continuing the review let me double check a couple of scenarios and update the testing instructions.

**Comment by monsieur-z** [comment]
Code and instructions updated 👌

File: client/components/jetpack/upsell-switch/index.tsx [hunk 1]
```diff
@@ -34,53 +35,66 @@ type Props = {
...
 	getStateForSite: QueryFunction;
+	isRequestingForSite: RequestFunction;
 	siteId: number | null;
 	siteState: SiteState | null;
 	atomicSite: boolean;
+	isRequesting: boolean;
 };
 
+const UI_STATE_LOADING = Symbol();
+const UI_STATE_LOADED = Symbol();
```
Comment (ChaosExAnima) [inline client/components/jetpack/upsell-switch/index.tsx:46]: This is a really cool use case for `Symbol`! One thing I'd wonder about is the overhead of transpiling to browsers that don't support it. It'd be easy to substitute strings, but not a huge deal either way.

**Comment by cleacos** [comment]
I'm going to merge and deploy it since the users are noticing it

File: client/components/jetpack/upsell-switch/index.tsx (hunk 1 above)
Comment (monsieur-z) [reply client/components/jetpack/upsell-switch/index.tsx:46]: True! I've seen it used in this way in other parts of the codebase but the polyfill won't work for more advanced features of Symbol indeed.

**Pull request merged**